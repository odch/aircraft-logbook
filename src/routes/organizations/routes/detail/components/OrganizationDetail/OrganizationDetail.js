import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import isLoaded from '../../../../../../util/isLoaded'
import {
  organization as organizationShape,
  aircraft as aircraftShape
} from '../../../../../../shapes'
import LoadingIcon from '../../../../../../components/LoadingIcon'
import AircraftList from '../AircraftList'

const styles = theme => ({
  container: {
    padding: '1em',
    [theme.breakpoints.up(1000 + theme.spacing(3 * 2))]: {
      width: 1000,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  notFoundMessage: {
    textAlign: 'center',
    marginTop: '1em'
  }
})

class OrganizationDetail extends React.Component {
  componentDidMount() {
    if (this.props.organization) {
      this.props.fetchAircrafts(this.props.organization.id)
    }
  }

  componentDidUpdate(prevProps) {
    if (
      this.props.organization &&
      (!prevProps.organization ||
        prevProps.organization.id !== this.props.organization.id)
    ) {
      this.props.fetchAircrafts(this.props.organization.id)
    }
  }

  render() {
    const {
      organizationId,
      userEmail,
      organization,
      aircrafts,
      classes
    } = this.props

    if (!isLoaded(organization)) {
      return <LoadingIcon />
    }

    if (!organization) {
      return this.renderContent(
        <div className={classes.notFoundMessage}>
          <FormattedMessage
            id="organization.notfound"
            values={{
              organizationId,
              userEmail,
              strong: chunks => <strong>{chunks}</strong>
            }}
          />
        </div>
      )
    }

    if (!isLoaded(aircrafts)) {
      return <LoadingIcon />
    }

    return this.renderContent(
      <>
        <Typography variant="h4" gutterBottom>
          {organizationId}
        </Typography>
        <AircraftList organization={organization} aircrafts={aircrafts} />
      </>
    )
  }

  renderContent(content) {
    const { classes } = this.props
    return <div className={classes.container}>{content}</div>
  }
}

OrganizationDetail.propTypes = {
  organizationId: PropTypes.string.isRequired,
  userEmail: PropTypes.string.isRequired,
  organization: organizationShape,
  aircrafts: PropTypes.arrayOf(aircraftShape),
  classes: PropTypes.object.isRequired,
  fetchAircrafts: PropTypes.func.isRequired
}

export default withStyles(styles)(OrganizationDetail)
