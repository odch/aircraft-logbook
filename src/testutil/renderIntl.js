import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
