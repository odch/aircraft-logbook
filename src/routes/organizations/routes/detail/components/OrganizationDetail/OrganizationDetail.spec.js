import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import renderIntl from '../../../../../../testutil/renderIntl'
import OrganizationDetail from './OrganizationDetail'

const StartPage = () => <div>Start Page</div>

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('detail', () => {
        describe('components', () => {
          describe('OrganizationDetail', () => {
            it('renders loading icon if organization not loaded', () => {
              const tree = renderIntl(
                <OrganizationDetail
                  organization={undefined}
                  fetchAircrafts={() => {}}
                />
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('renders loading icon if aircrafts not loaded', () => {
              const tree = renderIntl(
                <OrganizationDetail
                  organization={{ id: 'my_org' }}
                  aircrafts={undefined}
                  fetchAircrafts={() => {}}
                />
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('redirects to start page if organization was not found', () => {
              const tree = renderIntl(
                <Router>
                  <Switch>
                    <Route exact path="/" component={StartPage} />
                    <OrganizationDetail
                      organization={null}
                      fetchAircrafts={() => {}}
                    />
                  </Switch>
                </Router>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('renders organization if organizatino and aircrafts loaded', () => {
              const tree = renderIntl(
                <Router>
                  <OrganizationDetail
                    organization={{ id: 'my_org' }}
                    aircrafts={[
                      {
                        id: 'o7flC7jw8jmkOfWo8oyA',
                        registration: 'HBKFW'
                      },
                      {
                        id: 'BKi7HYAIoe1i75H3LMk1',
                        registration: 'HBKOF'
                      }
                    ]}
                    fetchAircrafts={() => {}}
                  />
                </Router>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('calls fetchAircrafts when mounted with organization', () => {
              const fetchAircrafts = jest.fn()

              renderIntl(
                <OrganizationDetail
                  organization={{ id: 'my_org' }}
                  aircrafts={undefined}
                  fetchAircrafts={fetchAircrafts}
                />
              )

              expect(fetchAircrafts).toBeCalledWith('my_org')
            })

            it('does not call fetchAircrafts when mounted without organization', () => {
              const fetchAircrafts = jest.fn()

              renderIntl(
                <OrganizationDetail
                  organization={undefined}
                  aircrafts={undefined}
                  fetchAircrafts={fetchAircrafts}
                />
              )

              expect(fetchAircrafts).not.toBeCalled()
            })

            it('calls fetchAircrafts when updated with new organization', () => {
              const fetchAircrafts = jest.fn()

              const renderer = renderIntl(
                <OrganizationDetail
                  organization={undefined}
                  aircrafts={undefined}
                  fetchAircrafts={fetchAircrafts}
                />
              )

              expect(fetchAircrafts).not.toBeCalled()

              renderer.update(
                <OrganizationDetail
                  organization={{ id: 'my_org' }}
                  aircrafts={undefined}
                  fetchAircrafts={fetchAircrafts}
                />
              )

              expect(fetchAircrafts).toBeCalledWith('my_org')
            })
          })
        })
      })
    })
  })
})
