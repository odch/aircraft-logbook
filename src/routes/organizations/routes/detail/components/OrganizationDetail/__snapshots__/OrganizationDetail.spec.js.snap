// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes detail components OrganizationDetail renders loading icon if aircrafts not loaded 1`] = `
"<div class=\\"LoadingIcon-container-3\\">
  <div class=\\"MuiCircularProgress-root LoadingIcon-icon-4 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width: 40px; height: 40px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\">
      <circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle>
    </svg></div>
</div>"
`;

exports[`routes organizations routes detail components OrganizationDetail renders loading icon if organization not loaded 1`] = `
"<div class=\\"LoadingIcon-container-3\\">
  <div class=\\"MuiCircularProgress-root LoadingIcon-icon-4 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width: 40px; height: 40px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\">
      <circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle>
    </svg></div>
</div>"
`;

exports[`routes organizations routes detail components OrganizationDetail renders organization if organization and aircrafts loaded 1`] = `
"<div class=\\"OrganizationDetail-container-1\\">
  <h4 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom\\"></h4>
  <div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3\\" data-cy=\\"aircraft-list\\">
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
      <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root AircraftList-card-47 MuiPaper-rounded\\" data-cy=\\"aircraft-card-o7flC7jw8jmkOfWo8oyA\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root AircraftList-actionArea-49\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA\\">
          <div class=\\"MuiCardContent-root AircraftList-cardContent-48\\">
            <h6 class=\\"MuiTypography-root MuiTypography-h6\\">HBKFW</h6>
          </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
        </a></div>
    </div>
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
      <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root AircraftList-card-47 MuiPaper-rounded\\" data-cy=\\"aircraft-card-BKi7HYAIoe1i75H3LMk1\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root AircraftList-actionArea-49\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/my_org/aircrafts/BKi7HYAIoe1i75H3LMk1\\">
          <div class=\\"MuiCardContent-root AircraftList-cardContent-48\\">
            <h6 class=\\"MuiTypography-root MuiTypography-h6\\">HBKOF</h6>
          </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
        </a></div>
    </div>
  </div>
</div>"
`;

exports[`routes organizations routes detail components OrganizationDetail renders with create button for aircrafts if is manager 1`] = `
"<div class=\\"OrganizationDetail-container-1\\">
  <h4 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom\\"></h4>
  <div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3\\" data-cy=\\"aircraft-list\\">
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
      <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root AircraftList-card-47 MuiPaper-rounded\\" data-cy=\\"aircraft-create-button\\"><button class=\\"MuiButtonBase-root MuiCardActionArea-root AircraftList-actionArea-49\\" tabindex=\\"0\\" type=\\"button\\">
          <div class=\\"MuiCardContent-root AircraftList-cardContent-48\\"><svg class=\\"MuiSvgIcon-root AircraftList-createIcon-50\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\">
              <path d=\\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z\\"></path>
            </svg><span class=\\"MuiTypography-root AircraftList-createText-51 MuiTypography-button\\">Neues Flugzeug erstellen</span></div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
        </button></div>
    </div>
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
      <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root AircraftList-card-47 MuiPaper-rounded\\" data-cy=\\"aircraft-card-o7flC7jw8jmkOfWo8oyA\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root AircraftList-actionArea-49\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA\\">
          <div class=\\"MuiCardContent-root AircraftList-cardContent-48\\">
            <h6 class=\\"MuiTypography-root MuiTypography-h6\\">HBKFW</h6>
          </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
        </a></div>
    </div>
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
      <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root AircraftList-card-47 MuiPaper-rounded\\" data-cy=\\"aircraft-card-BKi7HYAIoe1i75H3LMk1\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root AircraftList-actionArea-49\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/my_org/aircrafts/BKi7HYAIoe1i75H3LMk1\\">
          <div class=\\"MuiCardContent-root AircraftList-cardContent-48\\">
            <h6 class=\\"MuiTypography-root MuiTypography-h6\\">HBKOF</h6>
          </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
        </a></div>
    </div>
  </div>
</div>"
`;

exports[`routes organizations routes detail components OrganizationDetail renders without create button if aircrafts limit is reached 1`] = `
"<div class=\\"OrganizationDetail-container-1\\">
  <h4 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom\\"></h4>
  <div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3\\" data-cy=\\"aircraft-list\\">
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
      <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root AircraftList-card-47 MuiPaper-rounded\\" data-cy=\\"aircraft-card-o7flC7jw8jmkOfWo8oyA\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root AircraftList-actionArea-49\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA\\">
          <div class=\\"MuiCardContent-root AircraftList-cardContent-48\\">
            <h6 class=\\"MuiTypography-root MuiTypography-h6\\">HBKFW</h6>
          </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
        </a></div>
    </div>
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4\\">
      <div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root AircraftList-card-47 MuiPaper-rounded\\" data-cy=\\"aircraft-card-BKi7HYAIoe1i75H3LMk1\\"><a class=\\"MuiButtonBase-root MuiCardActionArea-root AircraftList-actionArea-49\\" tabindex=\\"0\\" role=\\"button\\" href=\\"/organizations/my_org/aircrafts/BKi7HYAIoe1i75H3LMk1\\">
          <div class=\\"MuiCardContent-root AircraftList-cardContent-48\\">
            <h6 class=\\"MuiTypography-root MuiTypography-h6\\">HBKOF</h6>
          </div><span class=\\"MuiCardActionArea-focusHighlight\\"></span><span class=\\"MuiTouchRipple-root\\"></span>
        </a></div>
    </div>
  </div>
</div>"
`;

exports[`routes organizations routes detail components OrganizationDetail shows message if organization was not found or user is not granted 1`] = `
"<div class=\\"OrganizationDetail-container-1\\">
  <div class=\\"OrganizationDetail-notFoundMessage-2\\">Die Organisation <strong>my_org</strong> wurde nicht gefunden oder dein Konto <strong>test@opendigital.ch</strong> ist nicht berechtigt, auf die Organisation zuzugreifen. Bitte wende dich an den Administrator der Organisation.</div>
</div>"
`;
